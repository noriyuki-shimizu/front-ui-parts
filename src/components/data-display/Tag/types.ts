/** Props */
export interface Props {
  /** カラー */
  color?: 'normal' | 'processing' | 'success' | 'error' | 'warning'
  /** 枠線 */
  bordered?: boolean
  /** ホバーした際に背景色を変更するか */
  hovered?: boolean
}
