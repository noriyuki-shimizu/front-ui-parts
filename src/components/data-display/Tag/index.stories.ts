import type { ArgTypeObject, Meta, StoryObj } from '@storybook/vue3'
import Tag from './index.vue'

const meta = {
  title: 'Data Display/Tag',
  // TODO: Watch: https://github.com/storybookjs/storybook/issues/24238
  component: Tag as Record<string, unknown>,
  tags: ['autodocs'],
  argTypes: {
    /* slots.default */
    default: {
      description: 'タグ内のテキスト',
      type: 'string',
      control: 'text'
    }
  },
  decorators: [() => ({ template: '<div style="display: flex; gap: 8px"><story /></div>' })]
} satisfies Meta<typeof Tag> | ArgTypeObject // TODO: Watch: https://github.com/storybookjs/storybook/issues/23352

export default meta

type Story = StoryObj<typeof meta>

export const Border: Story = {
  render: (args) => ({
    components: { Tag: Tag as Record<string, unknown> },
    setup() {
      return args
    },
    template: `
      <Tag bordered>Normal</Tag>
      <Tag bordered color="processing">Processing</Tag>
      <Tag bordered color="success">Success</Tag>
      <Tag bordered color="error">Error</Tag>
      <Tag bordered color="warning">Warning</Tag>
    `
  })
}

export const NoBorder: Story = {
  render: (args) => ({
    components: { Tag: Tag as Record<string, unknown> },
    setup() {
      return args
    },
    template: `
      <Tag>Normal</Tag>
      <Tag color="processing">Processing</Tag>
      <Tag color="success">Success</Tag>
      <Tag color="error">Error</Tag>
      <Tag color="warning">Warning</Tag>
    `
  })
}

export const Hover: Story = {
  render: (args) => ({
    components: { Tag: Tag as Record<string, unknown> },
    setup() {
      return args
    },
    template: `
      <Tag hovered>Normal</Tag>
      <Tag hovered color="processing">Processing</Tag>
      <Tag hovered color="success">Success</Tag>
      <Tag hovered color="error">Error</Tag>
      <Tag hovered color="warning">Warning</Tag>
    `
  })
}
