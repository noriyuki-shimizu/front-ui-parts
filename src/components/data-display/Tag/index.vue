<script lang="ts">
export default {
  name: 'Tag'
}
</script>

<script setup lang="ts">
import type { Props } from './types'

/** Props */
const props = withDefaults(defineProps<Props>(), {
  color: 'normal',
  bordered: false,
  hovered: false
})
</script>

<template>
  <span
    :class="[
      $style['tag'],
      $style[`tag--color-${props.color}`],
      { [$style['tag--border']]: props.bordered, [$style['tag--hover']]: props.hovered }
    ]"
  >
    <slot />
  </span>
</template>

<style module lang="scss">
@use './style.scss';
</style>
