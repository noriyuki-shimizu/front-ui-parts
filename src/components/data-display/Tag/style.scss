.tag {
  $this: &;

  padding: 8px;
  font-size: 12px;
  line-height: 1;
  border-radius: 8px;

  &--hover {
    &:hover {
      cursor: pointer;
    }
  }

  &--border {
    padding: 7px;
  }

  &--color {
    &-normal {
      background-color: var(--color-background-normal);

      &#{$this}--hover:hover {
        background-color: var(--color-background-normal-hover);
      }

      &#{$this}--border {
        border: 1px solid var(--color-border-normal);
      }
    }

    &-processing {
      color: var(--color-light-text-processing);
      background-color: var(--color-background-processing);

      &#{$this}--hover:hover {
        background-color: var(--color-background-processing-hover);
      }

      &#{$this}--border {
        border: 1px solid var(--color-border-processing);
      }
    }

    &-success {
      color: var(--color-light-text-success);
      background-color: var(--color-background-success);

      &#{$this}--hover:hover {
        background-color: var(--color-background-success-hover);
      }

      &#{$this}--border {
        border: 1px solid var(--color-border-success);
      }
    }

    &-error {
      color: var(--color-light-text-error);
      background-color: var(--color-background-error);

      &#{$this}--hover:hover {
        background-color: var(--color-background-error-hover);
      }

      &#{$this}--border {
        border: 1px solid var(--color-border-error);
      }
    }

    &-warning {
      color: var(--color-light-text-warning);
      background-color: var(--color-background-warning);

      &#{$this}--hover:hover {
        background-color: var(--color-background-warning-hover);
      }

      &#{$this}--border {
        border: 1px solid var(--color-border-warning);
      }
    }
  }
}
