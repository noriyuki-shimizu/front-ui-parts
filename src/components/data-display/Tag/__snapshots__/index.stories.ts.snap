// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Display/Tag Border smoke-test 1`] = `
<div style="display: flex; gap: 8px;">
  <span class="_tag_fo9cs_325 _tag--color-normal_fo9cs_337 _tag--border_fo9cs_334">
    Normal
  </span>
  <span class="_tag_fo9cs_325 _tag--color-processing_fo9cs_346 _tag--border_fo9cs_334">
    Processing
  </span>
  <span class="_tag_fo9cs_325 _tag--color-success_fo9cs_356 _tag--border_fo9cs_334">
    Success
  </span>
  <span class="_tag_fo9cs_325 _tag--color-error_fo9cs_366 _tag--border_fo9cs_334">
    Error
  </span>
  <span class="_tag_fo9cs_325 _tag--color-warning_fo9cs_376 _tag--border_fo9cs_334">
    Warning
  </span>
</div>
`;

exports[`Data Display/Tag Hover smoke-test 1`] = `
<div style="display: flex; gap: 8px;">
  <span class="_tag_fo9cs_325 _tag--color-normal_fo9cs_337 _tag--hover_fo9cs_331">
    Normal
  </span>
  <span class="_tag_fo9cs_325 _tag--color-processing_fo9cs_346 _tag--hover_fo9cs_331">
    Processing
  </span>
  <span class="_tag_fo9cs_325 _tag--color-success_fo9cs_356 _tag--hover_fo9cs_331">
    Success
  </span>
  <span class="_tag_fo9cs_325 _tag--color-error_fo9cs_366 _tag--hover_fo9cs_331">
    Error
  </span>
  <span class="_tag_fo9cs_325 _tag--color-warning_fo9cs_376 _tag--hover_fo9cs_331">
    Warning
  </span>
</div>
`;

exports[`Data Display/Tag NoBorder smoke-test 1`] = `
<div style="display: flex; gap: 8px;">
  <span class="_tag_fo9cs_325 _tag--color-normal_fo9cs_337">
    Normal
  </span>
  <span class="_tag_fo9cs_325 _tag--color-processing_fo9cs_346">
    Processing
  </span>
  <span class="_tag_fo9cs_325 _tag--color-success_fo9cs_356">
    Success
  </span>
  <span class="_tag_fo9cs_325 _tag--color-error_fo9cs_366">
    Error
  </span>
  <span class="_tag_fo9cs_325 _tag--color-warning_fo9cs_376">
    Warning
  </span>
</div>
`;
